consul/src/main/java/com/linecorp/armeria/internal/consul/AgentServiceClient.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.consul;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.common.QueryParams;
+
+/**
+ * Consul API client to register, deregister service and get list of services.
+ * {@code AgentServiceClient} is responsible for endpoint of Consul API:
+ * {@code `/agent/service`}(https://www.consul.io/api/agent/service.html)
+ *
+ * <p>GET /agent/services
+ * GET /agent/service/:service_id
+ * PUT /agent/service/register
+ * PUT /agent/service/deregister/:service_id
+ */
+final class AgentServiceClient {
+
+    private final Logger logger = LoggerFactory.getLogger(AgentServiceClient.class);
+    private final ConsulClient client;
+    private final ObjectMapper mapper;
+
+    /**
+     * Builds an AgentServiceClient with a ConsulClient.
+     */
+    public static AgentServiceClient of(ConsulClient consulClient) {
+        return new AgentServiceClient(consulClient);
+    }
+
+    private AgentServiceClient(ConsulClient client) {
+        this.client = client;
+        mapper = client.getObjectMapper();
+    }
+
+    /**
+     * Gets a list of services from consul agent.
+     */
+    public CompletableFuture<Map<String, Service>> services(
```

Comment:
I don't think this method is used.

Reply:
It used on the test. I will just remove the `public`.
How about that?

Reply:
SGTM